### PR TITLE
Add JSON output option to the "config home" subcommand

### LINKED ISSUE #5943: [feature] JSON output for `conan config home`
The command `conan config home` returns the directory where the config resides. Within our CI we use this to clean up before doing `conan config install ...`. Right now we parse the output of that command, but this could be errorprone as outputs sometimes change or are extended with additional information.

This feature request ist about adding a `--json` parameter like all the other commands have.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by theag3nt**
Changelog: Feature: JSON output to the "config home" subcommand with ``--json`` argument.
Docs: omit

As the issue referred to making command output more future-proof it can be written to file in JSON format (with the actual path contained in the `home` attribute).

Tests have been extended to make sure that the expected output is contained in the JSON file created.

Fixes #5943

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by theag3nt** [comment]
Based on the latest comment (https://github.com/conan-io/conan/issues/5943#issuecomment-544219806) on the issue I changed the behavior to only output a simple string into JSON.

**Comment by theag3nt** [comment]
Sure, I'll just revert to the first version I had before @monsdar's comment. :) Not sure that's reusable enough for a future `config`-wide JSON output, but exactly matches @lasote's request. I even added an assertion helper preparing for a future expansion of the `config home --json` output, so the test is checking that it contains `{"home": "<path>"}` as a subset.

You'll probably want to squash this to keep the history clean.

File: conans/test/functional/command/config_test.py [hunk 1]
```diff
@@ -98,16 +99,25 @@ def missing_subarguments_test(self):
...
+                                 json.loads(self.client.load("home.json")))
 
     def test_config_home_custom_home_dir(self):
         cache_folder = os.path.join(temp_folder(), "custom")
         with environment_append({"CONAN_USER_HOME": cache_folder}):
             client = TestClient(cache_folder=cache_folder)
             client.run("config home")
             self.assertIn(cache_folder, client.out)
+            client.run("config home --json home.json")
+            self._assert_dict_subset({"home": cache_folder}, json.loads(client.load("home.json")))
```
Comment (lasote) [inline conans/test/functional/command/config_test.py]: Why is this necessary?

Comment (lasote) [reply conans/test/functional/command/config_test.py]: `self.assertEqual` should work
